### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by ezraermy**
In this pull request, I: 
- Created a Classroom class with label and students attributes, along with getter and setter methods for label. 
- Created the Book class to include rentals attribute and an add_rental method. 
- Created the Rental class to include date, book, and person attributes. 
- Updated the Person class to include rentals attribute and an add_rental method.
- Modified the Student class to call the add_student method in its constructor, adding the student to the associated classroom.
### Other requirements
- No linter errors
- Follow github flow
- Document professional way.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label, :students
```
Comment (mikethreels) [inline classroom.rb]: - [ ] There is no need to change the `student` instance variable from outside the Classroom. therefor please use the attribute reader for this variable.

File: student.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 require_relative 'person'
 
 class Student < Person
+  attr_accessor :classroom
```
Comment (mikethreels) [inline student.rb]: - [ ] There is no need to change the `classroom` instance variable from outside the Student class. therefor please use the attribute reader for this variable.

File: student.rb [hunk 3]
```diff
@@ -1,6 +1,8 @@
...
+
```
Comment (mikethreels) [inline student.rb:5]: - [ ] In the student class there should be a method that serves as the 'belongs-to' side (a student belongs to a classroom). please have a look at [this article ](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships)with examples

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+  end
```
Comment (mikethreels) [inline book.rb]: - [ ] The method for adding rentals should look a little different than this, please have a look at[ this example](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) in a previous lesson

File: person.rb [hunk 5]
```diff
@@ -24,4 +25,8 @@ def correct_name
   def of_age?
     @age >= 18
   end
+
+  def add_rental(rental)
+    @rentals << rental
+  end
```
Comment (mikethreels) [inline person.rb]: - [ ] The method for adding rentals should look a little different than this, please have a look at[ this example](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) in a previous lesson

File: rental.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_accessor :date, :book, :person
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+
+    @book.add_rental(self)
+    @person.add_rental(self)
```
Comment (mikethreels) [inline rental.rb]: - [ ] int the constructor of the Rental class The rental should add itself to the @book and the @person achieving the 'belongs-to' association, please have a loot at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship)

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Ezra, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job making a start on the associations.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ezraermy** [comment]
Hi @mikethreels, thank you for your detailed and informative feedback.

**Comment by ezraermy** [comment]
Hi again @mikethreels. My microverse dashboard status says 'submit'. Do you have any idea why the status is not 'Change required'? Due to this reason, I couldn't pull a request for my second feedback.

**Comment by ezraermy** [comment]
Screenshot from 2023-05-24 22-11-40
My dashboard is not changing the status of the project into 'Change required'.

**Pull request merged**